BDArmory/Modules/MissileFire.cs

```diff
@@ -3486,12 +3603,17 @@ bool CheckEngagementEnvelope(IBDWeapon weaponCandidate, float distanceToTarget)
 
                         // lock radar if needed
                         if (ml.TargetingMode == MissileBase.TargetingModes.Radar)
-                            using (List<ModuleRadar>.Enumerator rd = radars.GetEnumerator())
-                                while (rd.MoveNext())
-                                {
-                                    if (rd.Current != null || rd.Current.canLock)
-                                        rd.Current.EnableRadar();
-                                }
+                        {
+                            List<ModuleRadar>.Enumerator rd = radars.GetEnumerator();
+                            while (rd.MoveNext())
+                            {
+                                if (rd.Current == null) continue;
+                                if (!rd.Current.canLock) continue;
+                                rd.Current.EnableRadar();
+                                break;
+                            }
+                            rd.Dispose();
+                        }
```

Comment:
This undoes #658 (make AI turn on all radars instead of just the first one), is this intentional?

(Haven't reviewed the rest, just this really stood out.)